### PR TITLE
Hightlight move of custom targets

### PR DESCRIPTION
**Pull request by theotherjimmy**
@rene-aguirre Could you review?

Resolves https://github.com/ARMmbed/mbed-os/issues/4723

### CODE REVIEW

**Comment by rene-aguirre** [comment]
Cherry on top... Would be possible to include also information about the expected `custom_targets.json` location at root of project directory?

**Comment by theotherjimmy** [comment]
Of course!

**Comment by theotherjimmy** [comment]
done

**Comment by rene-aguirre** [comment]
Thanks, looks good. 👍

**Comment by theotherjimmy** [comment]
@AnotherButler could you edit?

**Comment by theotherjimmy** [comment]
Phew. I almost forgot a period...

File: docs/advanced/config_system.md [hunk 1]
```diff
@@ -9,6 +9,8 @@ The mbed configuration system customizes the compile time configuration of vario
 
 The configuration system gathers and interprets all the configurations defined in the source tree. The output of the configuration system is a list of macros that are automatically defined when compiling the code.
 
+<span class="notes">**Note:** In prior releases, the configuration system provided a method for adding custom targets. Custom target are handled as [mbed targets](mbed_targets.md) in a special file named `custom_targets.json` in the root of a project.</span>
```
Comment (AnotherButler) [inline docs/advanced/config_system.md]: "are handled" is passive voice. Could we rewrite this as "This release handles custom targets as [mbed targets](mbed_targets.md) in a special file named `custom-targets.json` in the root of a project." or something similar?

Comment (theotherjimmy) [inline docs/advanced/config_system.md]: I had considered that, but wanted to avoid using "this release".

Comment (AnotherButler) [inline docs/advanced/config_system.md]: Could we change it to "mbed OS 5" or "The code" or something?

Comment (theotherjimmy) [inline docs/advanced/config_system.md]: I'm thinking "The mbed OS tools" for consistency with other docs.

File: docs/advanced/config_system.md [hunk 2]
```diff
@@ -9,6 +9,8 @@ The mbed configuration system customizes the compile time configuration of vario
 
 The configuration system gathers and interprets all the configurations defined in the source tree. The output of the configuration system is a list of macros that are automatically defined when compiling the code.
 
+<span class="notes">**Note:** In prior releases, the configuration system provided a method for adding custom targets. The mbed OS tools treat custom targets in a special file named `custom_targets.json` in the root of a project as [mbed targets](mbed_targets.md).</span>
```
Comment (theotherjimmy) [inline docs/advanced/config_system.md]: The second sentence needs work.

**Comment by theotherjimmy** [comment]
@AnotherButler I rewrote the second sentence.

File: docs/advanced/config_system.md [hunk 3]
```diff
@@ -9,6 +9,8 @@ The mbed configuration system customizes the compile time configuration of vario
 
 The configuration system gathers and interprets all the configurations defined in the source tree. The output of the configuration system is a list of macros that are automatically defined when compiling the code.
 
+<span class="notes">**Note:** In prior releases, the configuration system provided a method for adding custom targets. The mbed OS tools look for custom targets in a file named `custom_targets.json` in the root of a project, and treat them the same as [mbed targets](mbed_targets.md).</span>
```
Comment (AnotherButler) [inline docs/advanced/config_system.md]: Significantly better. Please delete the comma after "project". Also, should we clarify "them" refers to "custom targets", or do you think it's OK?

Comment (AnotherButler) [inline docs/advanced/config_system.md]: Also, should we add "now" after "The mbed OS tools" to clarify that it's not how prior releases did it?

**Pull request merged**